### PR TITLE
Convert Julia simulation to C++ with a Python interface

### PR DESCRIPTION
**Pull request by r-barnes**
Requesting a pull to nickolasclarke:master from nickolasclarke:richard/python

Write a message for this pull request. The first block
of text is the title and the rest is the description.

Changes:

6c1a113 (Richard Barnes, 8 minutes ago)
   Dropped sim.jl and associated Julia files

17f9180 (Richard Barnes, 40 minutes ago)
   Model runs completely

9a47a3f (Richard Barnes, 23 hours ago)
   Data passes to model!

639a0fe (Richard Barnes, 2 days ago)
   Beginning interface work

add392b (Richard Barnes, 2 days ago)
   Initial steps switching to Python

### CODE REVIEW

File: src/dispatch_package/dispatch.cpp [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+  std::string repr(){
+    return std::string("<dispatch.TripInfo ")
+      + "trip_id="            +trip_id                           +", "
+      + "block_id="           +std::to_string(block_id)          +", "
+      + "start_arrival_time=" +std::to_string(start_arrival_time)+", "
+      + "start_stop_id="      +std::to_string(start_stop_id)     +", "
+      + "end_arrival_time="   +std::to_string(end_arrival_time)  +", "
+      + "end_stop_id="        +std::to_string(end_stop_id)       +", "
+      + "distance="           +std::to_string(distance)          +", "
+      + "bus_busy_start="     +std::to_string(bus_busy_start)    +", "
```
Comment (nickolasclarke) [inline src/dispatch_package/dispatch.cpp:101]: what are these busy cols?

File: src/dispatch_package/dispatch.cpp [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+    block_start->start_depot_id = start_depot.depot_id;
+    block_start->energy_left    = params.battery_cap_kwh - energy_depot_to_start;
+    block_start->bus_id         = 1;
+
+    //Run through all the trips in the block
+    auto prevtrip=block_start;
+    for(auto trip=block_start;trip!=block_end;trip++){
+      //TODO: Incorporate charging at the beginning of trip if there is a charger present
+      auto trip_energy = trip->distance * params.kwh_per_km;
```
Comment (nickolasclarke) [inline src/dispatch_package/dispatch.cpp:179]: perhaps we should log the full energy cycle of a block to evaluate for patterns that get us in the energy trap?

**Review by nickolasclarke — APPROVED** [review]
the the cpp is harder for me to grok, but no glaring errors

File: src/dispatch_package/dispatch.cpp (hunk 2 above)
Comment (r-barnes) [reply src/dispatch_package/dispatch.cpp:179]: Let's go over this logic with a fine-toothed comb shortly.

**Pull request merged**